app/views/CreateDiscussionView/index.tsx

```diff
@@ -95,117 +98,89 @@ class CreateDiscussionView extends React.Component<ICreateChannelViewProps, ICre
 			reply,
 			users
 		};
-		if (this.isEncryptionEnabled) {
+		if (isEncryptionEnabled) {
 			params.encrypted = encrypted ?? false;
 		}
 
 		dispatch(createDiscussionRequest(params));
 	};
 
-	valid = () => {
-		const { channel, name } = this.state;
-
-		return channel && channel.rid && channel.rid.trim().length && name?.trim().length;
-	};
-
-	selectChannel = ({ value }: { value: ISearchLocal }) => {
-		logEvent(events.CD_SELECT_CHANNEL);
-		this.setState({ channel: value, encrypted: value?.encrypted });
-	};
-
-	selectUsers = ({ value }: { value: string[] }) => {
-		logEvent(events.CD_SELECT_USERS);
-		this.setState({ users: value });
-	};
-
-	get isEncryptionEnabled() {
-		const { channel } = this.state;
-		const { encryptionEnabled } = this.props;
-		return encryptionEnabled && E2E_ROOM_TYPES[channel?.t];
-	}
+	useEffect(() => {
+		if (loading === prevLoading) {
```

Comment:
I'm the one who suggested the usage of `usePrevious`, but I don't think it's a good approach now 😬 
`usePrevious` uses a useEffect based on the param and keeps the value on a ref.

Since the previous value is being used inside another useEffect here, I think we should remove the usage of `usePrevious` and use a regular ref instead.

`usePrevious` is useful when we don't have/want to flood the component with another lifecycle.